jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js

```diff
@@ -277,7 +277,7 @@ YUI.add('deployment-credential-add', function() {
                   regex: /\S+/,
                   error: 'This field is required.'
                 }, {
-                  regex: /^[a-zA-Z0-9]([a-zA-Z0-9.-]*[a-zA-Z0-9])?$/,
+                  regex: /^([a-zA-Z0-9]([a-zA-Z0-9.-]*[a-zA-Z0-9])?)?$/,
```

Comment:
IIRC this regex was grabbed from the Juju source - Can you confirm that this is still correct?